frodo.js

```diff
@@ -6,40 +6,34 @@
  *
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.